io_u.c

```diff
@@ -884,6 +883,23 @@ static int fill_io_u(struct thread_data *td, struct io_u *io_u)
 	}
 
 	/*
+	 * If zone_size > zone_range, then maintain the same zone until
+	 * zone_bytes >= zone_size.
+	 */
+	if (f->last_pos[io_u->ddir] >= (f->file_offset + td->o.zone_range)) {
+		dprint(FD_IO, "io_u maintain zone offset=%" PRIu64 "/last_pos=%" PRIu64 "\n",
+				f->file_offset, f->last_pos[io_u->ddir]);
+		f->last_pos[io_u->ddir] = f->file_offset;
+	}
+
+	/*
+	 * For random: if 'norandommap' is not set and zone_size > zone_range,
+	 * map needs to be reset as it's done with zone_range everytime.
+	 */
+	if (td->zone_bytes % td->o.zone_range == 0)
```

Comment:
What happens if td->o.zone_range == 0?

You also might want to consider using parenthesis to make this clearer to read:

```if ((td->zone_bytes % td->o.zone_range) == 0)```

but the mod-by-0 is the biggest issue here.

Reply:
makes sense. I have made changes and updated the tree with it. Please let me know. Thanks.